clang/lib/Driver/ToolChains/HIPSPV.cpp

```diff
@@ -70,6 +70,15 @@ void HIPSPV::Linker::constructLinkAndEmitSpirvCommand(
 
   // Link LLVM bitcode.
   ArgStringList LinkArgs{};
+
+  // Add static device libraries using the common helper function.
+  // This handles unbundling archives (.a) containing bitcode bundles.
+  const HIPSPVToolChain &TC = static_cast<const HIPSPVToolChain &>(getToolChain());
+  StringRef Arch = TC.getTriple().getArchName();
```

Comment:
Shouldn't need casting as the base ToolChain has `getTriple()` method.
```suggestion
  StringRef Arch = getToolChain().getTriple().getArchName();
```